### PR TITLE
Create formula of jumanpp, RNNLM based Japanese morphological analyzer

### PR DESCRIPTION
**Pull request by chezou**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

I created new formula of jumanpp, which is Japanese morphological analyzer based on deep learning.

### CODE REVIEW

File: Formula/jumanpp.rb
Comment (MikeMcQuaid) [inline Formula/jumanpp.rb]: Can you please submit this issue to the upstream developers of this project and add a link to the upstream submission and explanation of why `ENV.deparallelize` (instead of `-j1`) is needed in a comment in the formula file. Thanks!

File: Formula/jumanpp.rb
Comment (MikeMcQuaid) [inline Formula/jumanpp.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Additionally, use `system "#{bin}/jumanpp"`. Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/jumanpp.rb
Comment (chezou) [inline Formula/jumanpp.rb]: Unfortunately, there is no public repo or issue board currently. I reported on twitter.
https://twitter.com/chezou/status/786687092034240512
Would it be possible to write this tweet link?

File: Formula/jumanpp.rb
Comment (chezou) [inline Formula/jumanpp.rb]: fixed f488229

File: Formula/jumanpp.rb
Comment (chezou) [inline Formula/jumanpp.rb]: Finally, I found out the problem and create patch. https://github.com/Homebrew/homebrew-core/pull/5875/commits/0e079aaebc9bbdb57f9610692ab8de266b79a06c
Is is OK?

Comment (bfontaine) [inline Formula/jumanpp.rb]: We can’t accept a patch that upstream won’t merge. The project homepage lists Hajime Morita, Daisuke Kawahara and Sadao Kurohashi as authors; you can find their email addresses here: http://nlp.ist.i.kyoto-u.ac.jp/EN/index.php?Members

Comment (chezou) [inline Formula/jumanpp.rb]: @bfontaine  I've reported the patch to Hajime Morita via Twitter https://twitter.com/chezou/status/786753665671442433
Should I have to wait until patch will be merged?

Comment (chezou) [inline Formula/jumanpp.rb]: According to the Cook-book, it is required patches to be **submit**, not to be **merged**.

> If not, file a bug report and/or submit your patch for inclusion. We may sometimes still accept your patch before it was submitted upstream but by getting the ball rolling on fixing the upstream issue you reduce the length of time we have to carry the patch around.

And as I wrote, I already [submitted the patch to author on twitter](https://twitter.com/chezou/status/786753665671442433), or is there any rule that all softwares to be proposed to homebrew should have public issues? Unfortunately, they don't have any public repo at present.

Comment (MikeMcQuaid) [inline Formula/jumanpp.rb]: They don't need to have public issues but for new formulae we are more conservative on patches because we need to see if the upstream project wishes to support macOS at all.

File: Formula/jumanpp.rb
Comment (dunn) [inline Formula/jumanpp.rb]: This patch needs to be checksummed; add a `sha256` line below the `url`.

File: Formula/jumanpp.rb
Comment (chezou) [inline Formula/jumanpp.rb]: Now the repo is opened, and [fixed this problem on upstream](https://github.com/ku-nlp/jumanpp/commit/4cabe0fb0ff28d0176a11f40b959e224eccc541e)

**Comment by dunn** [comment]
Thanks! Could you squash these down into a single commit?

**Comment by dunn** [comment]
Thanks! @MikeMcQuaid, any last thoughts?

File: Formula/jumanpp.rb
Comment (MikeMcQuaid) [inline Formula/jumanpp.rb]: `pipe_output` should work for this, I think. Would be good to `assert_match` on the output, too.

File: Formula/jumanpp.rb
Comment (MikeMcQuaid) [inline Formula/jumanpp.rb]: Please add a Ruby comment stating where and when this patch was submitted upstream, thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/jumanpp.rb
Comment (chezou) [inline Formula/jumanpp.rb]: fixed f3a27de

**Comment by chezou** [comment]
I added a comment for patch https://github.com/Homebrew/homebrew-core/pull/5875/commits/d1bb43f6c7428a6adb78a0a10c9ff8d6cad397d1
and used `pipe_output` https://github.com/Homebrew/homebrew-core/pull/5875/commits/f3a27de611c79c383e9e0448d7a0ff82c2beb925
If OK, I'll squash again.

**Comment by chezou** [comment]
Are there still any problems?

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by chezou** [comment]
Thanks for approval! I've just squashed.

**Comment by MikeMcQuaid** [comment]
It's alright, I'll do that locally.

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `jumanpp 1.01 (new formula)`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Comment by chezou** [comment]
Thanks!

**Pull request closed without merging**
